types/Schemas.ts

```diff
@@ -40,3 +40,25 @@ export type CreateObjectsResponse = {
     archived: boolean;
   }>;
 };
+
+export type SchemaDefinition = {
+  allowsSensitiveProperties?: boolean;
+  associatedObjects?: Array<string>;
+  description?: string;
+  labels: {
+    plural: string;
+    singular: string;
+  };
+  searchableProperties?: Array<string>;
+  secondaryDisplayProperties?: Array<string>;
+  name: string;
+  primaryDisplayProperty?: string;
+  properties: [
+    {
+      isPrimaryDisplayLabel: true;
+      label: string;
+      name: string;
+    },
+  ];
+  requiredProperties: Array<string>;
+};
```

Comment:
Non-blocking: it would probably make sense to keep all custom object related types in one file. Ideally that would be `CustomObjects.ts`, but that would be a breaking change. We should either put everything in `Schemas.ts` or make a note to fix this during the next major release